### DISCUSSION TITLE
DAI 2023-2024 - Practical work 1 - Aurélien Richard

### DISCUSSION THREAD

**Discussion by Lyrafll** [heig-vd-dai-course/heig-vd-dai-course#89] [Show and tell]
This first practical work will serve as an introduction to proper git usage and Java IOs.
In order to do so we are tasked to create a simple CLI (basic + more functionalities if wanted) using Maven, Java IOs and all other tools at our disposal.

**Comment by ludelafo** [comment]
Great! Could you tell us a bit more about the idea you have for the CLI tool?

It is OK if you want to replicate the demo made by Hadrien in class if you want. If you do so, we will ask you to add a few more commands as the CLI presented by Hadrien is a fairly simple one.

**Reply by Lyrafll** [reply]
Sure !
If I understood well we need to put a twist to it right ?
I was thinking about using it for """"encryption""" using the [Caesar cipher](https://en.wikipedia.org/wiki/Caesar_cipher).
If we can add commands in the cli I would like to add the ability to choose by how much we want to shift the letters.
So for example, we would use -s x (where x is the range of the shift. eg: if x = 2 then A -> C) to let the user choose and by default I can set the shift to 3.

The output file would be the result of the shift 👍

**Reply by ludelafo** [reply]
Seems perfect! If you have time, you could add two commands, `encrypt` and `decrypt`, to go both ways!

**Comment by Lyrafll** [comment]
https://github.com/Lyrafll/DAI-Practical-Work-1
commit : 73b9007bd1922221bf418252a76545620892ea79
